heartbeat/monitors/active/http/checkjson.go

```diff
@@ -0,0 +1,175 @@
+				matches, err := eval.EvalBool(context.Background(), d)
+				if err != nil {
+					// Conditions cannot match array root JSON responses
+					return false
+				}
+
+				return matches
+			}
+
+			expressionChecks = append(expressionChecks, compiledJsonCheck{
+				description: check.Description,
+				check:       checkFn,
+				source:      check.Expression,
+			})
+		} else if check.Condition != nil {
+			cfgwarn.Deprecate("8.0.0", "JSON conditions are deprecated, use 'expression' instead.")
+			cond, err := conditions.NewCondition(check.Condition)
+			if err != nil {
+				return nil, fmt.Errorf("could not load JSON condition '%s': %w", check.Description, err)
+			}
+
+			checkFn := func(d interface{}) bool {
+				ms, ok := d.(map[string]interface{})
+				if ok {
+					return cond.Check(common.MapStr(ms))
+				} else {
+					return false
+				}
+			}
+			conditionChecks = append(conditionChecks, compiledJsonCheck{
+				description: check.Description,
+				check:       checkFn,
+				source:      fmt.Sprintf("%v", check.Condition),
+			})
+		}
+	}
+
+	return createJsonCheck(expressionChecks, conditionChecks), nil
+}
+
+func createJsonCheck(expressionChecks []compiledJsonCheck, conditionChecks []compiledJsonCheck) bodyValidator {
+	return func(_ *http.Response, body string) error {
+		var validationFailures []string
+		if len(expressionChecks) > 0 {
+			decoded, err := decodeJson(body, false)
+			if err != nil {
+				return err
+			}
+			validationFailures = append(validationFailures, runCompiledJSONChecks(decoded, expressionChecks)...)
+		}
+
+		if len(conditionChecks) > 0 {
+			decoded, err := decodeJson(body, true)
+			if err != nil {
+				// This should only err if the JSON is unparsable,
+				// so no need to handle returning 'errs' if this happens
+				return err
+			}
+			validationFailures = append(validationFailures, runCompiledJSONChecks(decoded, conditionChecks)...)
+		}
+
+		if len(validationFailures) > 0 {
+			bodyTrunc := len(body)
+			if bodyTrunc > 2048 {
+				bodyTrunc = 2048
+			}
+			return fmt.Errorf(
+				"JSON body did not match %d expressions or conditions '%s'. Received JSON (first 2048 chars): %s",
+				len(validationFailures),
+				strings.Join(validationFailures, ","),
+				body[:bodyTrunc-1], // Only print the first 2k of JSON to limit size
+			)
+		}
+
+		return nil
+	}
+}
+
+func decodeJson(body string, forCondition bool) (result interface{}, err error) {
+	decoder := json.NewDecoder(strings.NewReader(body))
+	// Condition checks need to convert the parsed numeric
+	// values in a way appropriate for the condition evaluator. GVal only works if
+	// this is not enabled, so expression checks have a separate codepath.
+	if forCondition {
+		decoder.UseNumber()
+	}
+
+	err = decoder.Decode(&result)
+	if err != nil {
+		return result, fmt.Errorf("could not parse JSON: %w", err)
+	}
+
+	if forCondition {
+		if resMap, ok := result.(map[string]interface{}); ok {
+			if forCondition {
```

Comment:
if condition is not relevant here as we have this block only for condition checks.